common/djangoapps/util/course.py

```diff
@@ -2,27 +2,47 @@
 Utility methods related to course
 """
 import logging
-from django.conf import settings
 
+from django.conf import settings
 from opaque_keys.edx.keys import CourseKey
 
+from openedx.core.djangoapps.catalog.utils import get_run_marketing_url
+
 log = logging.getLogger(__name__)
 
 
-def get_lms_link_for_about_page(course_key):
+def get_link_for_about_page(course_key, user):
     """
     Returns the url to the course about page.
     """
     assert isinstance(course_key, CourseKey)
 
     if settings.FEATURES.get('ENABLE_MKTG_SITE'):
-        # Root will be "https://www.edx.org". The complete URL will still not be exactly correct,
-        # but redirects exist from www.edx.org to get to the Drupal course about page URL.
-        about_base = settings.MKTG_URLS['ROOT']
-    else:
-        about_base = settings.LMS_ROOT_URL
+        marketing_url = get_run_marketing_url(course_key, user)
+        if marketing_url:
```

Comment:
The previous logic was - if marketing site is enabled return marketing url.
Otherwise return lms url.
The current logic is, if marketing site is enabled or marketing site url is null, return lms url.
Are you sure we don't want to put the last return statement in this function in an else?